### PR TITLE
Fix/384 step fixes

### PR DESCRIPTION
**Pull request by harshil1793**
### This PR fixes:
**Step 9**
- "Benefits Selection" should be "Plot or interment Allowance" (two text fixes at the top of the page)
- 2 occurrences to change from "cemetary" to "cemetery"

**Step 10-11**
- Claimant Information email not showing up on the review page even though data entry exists on the form page
- Format phone number better (333) 555-7777 in review page (check validation for non numeric and -)

## Original issue(s)
department-of-veterans-affairs/va-forms-system-core#384

## Screenshots
**Email on review page:**
email on review page
**Phone Format:**
phone
**Text typo fix:**
typo fix
**Plot title fix:**
plot
**Allowance title fix:**
allowance

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link to the original github issue has been provided
- [ ] No sensitive information (i.e. PII/credentials/internal URLs etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: examples/burial-form/BurialAllowance.jsx [hunk 1]
```diff
@@ -44,7 +44,7 @@ export default function BurialAllowance(props) {
 
   return (
     <>
-      <Page {...props}>
+      <Page {...props} title="Burial allowance">
```
Comment (mlauren) [inline examples/burial-form/BurialAllowance.jsx:47]: Title should be updated and passed through the route on the index page.

**Review by mlauren — CHANGES_REQUESTED** [review]
Besides the place where to update the title (in the routes on the index page), LGTM

**Review by mlauren — APPROVED** [review]

File: examples/burial-form/index.jsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const BurialApp = (props) => {
         <Route path="/military-history/previous-names" element={<PreviousNames title="Military history" />} />
         <Route path="/benefits/selection" element={<BenefitsSelection title="Benefits Selection" />} />
         <Route path="/benefits/burial-allowance" element={
-          <ConditionalRoute title="Benefits Selection: Burial Allowance" type="conditional" condition={'benefitsSelection.burialAllowance'}>
+          <ConditionalRoute title="Benefits Selection" type="conditional" condition={'benefitsSelection.burialAllowance'}>
```
Comment (bjmccotter7192) [inline examples/burial-form/index.jsx:47]: I think this should still be `Benefits Selection: Burial Allowance` and same for the `Benefits Selection: Plot Allowance`

File: examples/burial-form/PlotAllowance.jsx [hunk 3]
```diff
@@ -12,12 +12,12 @@ export default function PlotAllowance(props) {
   
   return (
     <div className="vads-u-margin-x--1p5">
-      <Page {...props}>
+      <Page {...props} title="Plot or interment allowance">
```
Comment (bjmccotter7192) [inline examples/burial-form/PlotAllowance.jsx:15]: Title should be passed in the index.jsx

**Review by bjmccotter7192 — APPROVED** [review]
LGTM just a couple things

File: examples/burial-form/BurialAllowance.jsx (hunk 1 above)
Comment (harshil1793) [reply examples/burial-form/BurialAllowance.jsx:47]: Actually the reason for this is it is an internal step of Burial selection. So, the title in progress bar would be "Burial selection" and the title for page would be "Burial allowance" according to the mockups.

**Pull request merged**